improver/categorical/utilities.py

```diff
@@ -241,19 +248,25 @@ def interrogate_decision_tree(decision_tree: Dict[str, Dict[str, Any]]) -> str:
     """
     # Diagnostic names and threshold values.
     requirements = {}
+    output = []
     for key, query in decision_tree.items():
         if not is_decision_node(key, query):
             continue
         diagnostics = get_parameter_names(
             expand_nested_lists(query, "diagnostic_fields")
         )
-        thresholds = expand_nested_lists(query, "diagnostic_thresholds")
-        for diagnostic, threshold in zip(diagnostics, thresholds):
-            requirements.setdefault(diagnostic, set()).add(threshold)
+        if "deterministic" in query:
```

Comment:
This would be true if the "deterministic" value were False. Replace this with the `get` method suggested elsewhere.

Reply:
Done